### ISSUE TITLE
Feature Request: Pluggable OAuth providers

### ISSUE THREAD

**Issue by georgehaidar** [concourse/concourse#539]
My company relies heavily on ADFS to manage developer access to various services such as AWS. We tend to gravitate towards services that integrate with ADFS as an authorization provider. Having said that, we recently started using ADFS 3.0 which exposes an OAuth2 provider. This now allows it to integrate to many more services.
ConcourseCI currently supports Github OAuth. Can it be extended to use any OAuth provider?

**Comment by HenryTheHamster** [comment]
We also have a strong requirement for ADFS. I understand the Concourse team is planning to make Teams first-class citizens, and this information is pulled from the OAuth provider - hence Github-specific OAuth instead of generic.
Would an OAuth solution that made the assumption that every user belonged to the same 'team' allow a generic OAuth provider (which we can use for ADFS), at the cost of not being able to manage different teams at the Concourse level?

**Comment by vito** [comment]
yep that sounds reasonable

**Comment by cppforlife** [comment]
Seems like UAA supports ADFS integration: https://github.com/cloudfoundry/uaa/blob/master/docs/UAA-Overview.rst

**Comment by vito** [comment]
(...which will be supported by Teams at launch)

**Comment by vito** [comment]
Actually nevermind; our UAA support at launch will be CF-specific as it'll rely on Cloud Controller for determining access. Maybe in a later iteration.

**Comment by georgehaidar** [comment]
@vito I wanted to clarify that by pluggable I meant configurable. I imagined something like the following would be simple to implement.

```
concourse web \
  --oauth-authorize-endpoint https://myprovider.com/oauth/authorize \
  --oauth-token-endpoint https://myprovider.com/oauth/access_token \
  --oauth-client-id 123456 \
  --oauth-client-secret abcdef
```

**Comment by vito** [comment]
@georgehaidar Yep that sounds reasonable. Should be relatively straightforward to add, aside from plumbing the flags through.

**Comment by devorbitus** [comment]
Isn't this already done

**Comment by vito** [comment]
yup! shipped with 2.0. forgot to close. thanks.
